app/src/main/cpp/native-lib.cpp

```diff
@@ -36,83 +30,175 @@ namespace
+                if (getEnvStat == JNI_EDETACHED) {
+                    std::cout << "GetEnv: not attached" << std::endl;
+                    if (jvm->AttachCurrentThread((JNIEnv**)&env, nullptr) != JNI_OK) {
+                        std::cout << "Failed to attach" << std::endl;
+                        return;
+                    }
+                } else if (getEnvStat == JNI_EVERSION) {
+                    std::cout << "GetEnv: version not supported" << std::endl;
+                    return;
+                }
+
+                jclass jcallback_class = env->GetObjectClass(this_ref);
+                jmethodID jcallback_method = env->GetMethodID(jcallback_class, "oepDataReady", "([BII)V");
+
+                auto size = image->get_width() * image->get_height() * image->get_bytes_per_pixel();
+                auto byte_array = env->NewByteArray(size);
+                env->SetByteArrayRegion(byte_array, 0, size, reinterpret_cast<const jbyte*>((uint8_t*)image->get_base_sptr().get()));
+
+                // call callback
+                env->CallVoidMethod(this_ref, jcallback_method, byte_array, image->get_width(), image->get_height());
+
+                if (env->ExceptionCheck()) {
+                    env->ExceptionDescribe();
+                }
+                env->DeleteGlobalRef(this_ref);
+
+                jvm->DetachCurrentThread();
+            };
+            // Get texture id from shared context and render it
+            result->get_image(bnb::oep::interfaces::image_format::bpc8_rgba, get_image_callback);
+        }
+    };
+    oep->process_image_async(pb_image, bnb::oep::interfaces::rotation::deg0, get_pixel_buffer_callback, bnb::oep::interfaces::rotation::deg180);
 }
 
-extern "C" JNIEXPORT jbyteArray JNICALL
-Java_com_banuba_sdk_example_quickstart_1cpp_BanubaSdk_processPhoto(JNIEnv* env, jobject thiz, jlong effect_player, jobject rgba, jint width, jint height)
+/* kotlin BanubaSdk::surfaceChanged */
+extern "C" JNIEXPORT void JNICALL
+Java_com_banuba_sdk_example_quickstart_1c_1api_BanubaSdk_surfaceChanged(JNIEnv* env, jobject thiz, jlong jsdk, jint jwidth, jint jheight)
 {
-    auto sdk = (BanubaSdkManager*) effect_player;
-
-    auto* data = static_cast<uint8_t*>(env->GetDirectBufferAddress(rgba));
-
-    bnb_image_format_t image_format{
-                static_cast<uint32_t>(width),
-                static_cast<uint32_t>(height),
-                bnb_image_orientation_t::BNB_DEG_0,
-                /* is_mirrored */ false,
-                /* face_orientation */ 0};
+    auto oep = reinterpret_cast<BanubaSdkManager*>(jsdk)->oep;
+    auto width = static_cast<int32_t>(jwidth);
+    auto height = static_cast<int32_t>(jheight);
+    oep->surface_changed(width, height);
+}
 
-    auto rgb_in = std::make_shared<rgb_image>(color_plane_weak(data), image_format, bnb_pixel_format_t::BNB_RGBA);
-    auto pb = sdk->oep->process_image(rgb_in);
-    auto rgba_out_opt = pb->get_rgba();
+/* kotlin BanubaSdk::loadEffect */
```

Comment:
What does it mean the word kotlin in the comment?

Reply:
This is just a note that the interface is from Kotlin